### PR TITLE
ebuild-maintenance/general: New section on maintaining ebuilds

### PR DESCRIPTION
**Pull request by thesamesam**
This is a draft new section on "Maintaining your package".

It intends to state the obvious (but possibly unwritten) rules we have around maintaining packages in Gentoo, such as checking for open bugs regularly, and keeping up with stabilisation once your package has received stable keywords.

Signed-off-by: Sam James <sam@gentoo.org>

### CODE REVIEW

File: ebuild-maintenance/general/text.xml
Comment (ajakk) [inline ebuild-maintenance/general/text.xml:43]: Maybe say something about the `;maintainer=` parameter?

File: ebuild-maintenance/general/text.xml
Comment (ajakk) [inline ebuild-maintenance/general/text.xml:58]: This is the devmanual, but nondevelopers often use it too, so it might be worth mentioning channels that don't have +m?

File: ebuild-maintenance/general/text.xml
Comment (ajakk) [inline ebuild-maintenance/general/text.xml:80]: the mailing list and CI messages

File: ebuild-maintenance/general/text.xml
Comment (ulm) [inline ebuild-maintenance/general/text.xml:64]:
```suggestion
see the <uri link="::keywording/#Moving from ~arch to arch">stabilization section</uri>),
provided there
```

File: ebuild-maintenance/general/text.xml
Comment (ulm) [inline ebuild-maintenance/general/text.xml:35]: I think this sentence is more distracting than it is helpful. Bugzilla has a powerful search function with the possibility to save searches, so why would anyone prefer p.g.o (which IIUC is aimed primarily at users) to Bugzilla itself?

File: ebuild-maintenance/general/text.xml
Comment (ulm) [inline ebuild-maintenance/general/text.xml:55]: Does mentioning of croaker help developers with ebuild maintenance? Using IRC is optional, and anyone present in #-dev will notice these messages.

File: ebuild-maintenance/general/text.xml
Comment (ulm) [inline ebuild-maintenance/general/text.xml:4]: Use _Title Case_ in headings.
```suggestion
<title>Maintaining Your Package</title>
```

File: ebuild-maintenance/general/text.xml
Comment (thesamesam) [reply ebuild-maintenance/general/text.xml:55]: Agreed, dropping.

File: ebuild-maintenance/general/text.xml
Comment (thesamesam) [reply ebuild-maintenance/general/text.xml:35]: Agreed, let's restore it if the functionality is complete in b.g.o. Note that the maintainer pages on p.g.o are designed for developers and feedback is welcome on how to make them better for that.

File: ebuild-maintenance/general/text.xml
Comment (thesamesam) [reply ebuild-maintenance/general/text.xml]: Very good point, thank you!

File: ebuild-maintenance/general/text.xml
Comment (sbraz) [inline ebuild-maintenance/general/text.xml:103]: I would move the "with":
>  e.g. Python versions with which your package is compatible as soon as possible

or drop it and write something like:
> e.g. Python versions which your package supports as soon as possible

File: ebuild-maintenance/general/text.xml
Comment (sbraz) [inline ebuild-maintenance/general/text.xml:105]: change**s**

File: ebuild-maintenance/general/text.xml
Comment (sbraz) [inline ebuild-maintenance/general/text.xml:55]: Could we also mention the per-maintainer URLs (`;maintainer=<name>`)?

File: ebuild-maintenance/general/text.xml
Comment (gktrk) [inline ebuild-maintenance/general/text.xml:67]: Citation needed :P I doubt anyway would refute the technical superiority of pkgcheck. Perhaps we should work around this controversy by simply stating that fact and telling that irrespective of the tool being used, it's the developer's responsibility to ensure quality control. This way we can keep this paragraph short and to the point.

File: ebuild-maintenance/general/text.xml
Comment (gktrk) [inline ebuild-maintenance/general/text.xml:88]: I think this is redundant. No developer with common sense should be arguing that not stabilizing their packages is better.

File: ebuild-maintenance/general/text.xml
Comment (gktrk) [inline ebuild-maintenance/general/text.xml:98]: If this is meant to be a separate paragraph, it should have its own `<p/>`. But I think we can meld this to the previous paragraph by simply saying `If you are blocked by a dependency not supporting compatible support variables, file a bug or contact the maintainer of the dependency to promptly resolve it.`

**Review by gktrk — CHANGES_REQUESTED** [review]
Without blocking the inclusion of this PR, we should start thinking about how this fits into ebuild-maintenance. Before, we had a single page that's similar to this in concept and we ended up creating more subsections eventually. The title doesn't quite fit IMO but like I said, it shouldn't block this PR.

File: ebuild-maintenance/general/text.xml
Comment (juippis) [inline ebuild-maintenance/general/text.xml:18]: If we are going to make this canon, I'd like this to be more precise:
```suggestion
You should always use the latest <uri link="::ebuild-writing/eapi/">EAPI</uri>
when adding new ebuilds or eclasses to ::gentoo repository.
```
Because it's fine to not bump EAPI when you add a patch, or eg a subslot binder to a dep with new subslot. Of course this is also common sense.

File: ebuild-maintenance/general/text.xml
Comment (juippis) [inline ebuild-maintenance/general/text.xml:24]: What does this mean? If EAPI-8 was pushed today, no EAPI-8 ebuild was going to be stabilized for a year? That doesn't seem right.

File: ebuild-maintenance/general/text.xml
Comment (juippis) [inline ebuild-maintenance/general/text.xml:37]: "Respond and resolve bugs assigned to you. If you do not use package assigned to you anymore or can't maintain it, pass it on to someone else." (with a link to https://devmanual.gentoo.org/general-concepts/package-maintainers/index.html#adding-and-removing-maintainers)

File: ebuild-maintenance/general/text.xml
Comment (juippis) [reply ebuild-maintenance/general/text.xml:52]: I feel like devmanual should have a section dedicated to pkgcheck, as it's hard to find documentation on how to use it properly.

File: ebuild-maintenance/general/text.xml
Comment (juippis) [inline ebuild-maintenance/general/text.xml:74]: This whole paragraph seems hypotethical and not fitting for devmanual? There's a lot of assumption and "thoughtflowing" involved. I'd rather have it replaced with concrete example how to run `repoman` and `pkcheck` parallel in everyday ebuild maintenance.

**Comment by ulm** [comment]
> Without blocking the inclusion of this PR, we should start thinking about how this fits into ebuild-maintenance. Before, we had a single page that's similar to this in concept and we ended up creating more subsections eventually. The title doesn't quite fit IMO but like I said, it shouldn't block this PR.

Pasting conversation from #gentoo-dev for reference:

&lt;@ulm&gt; sam_: gokturk has a point that a section with title "general" may indicate a structural problem. especially because ebuild-maintenance was sort of a catch-all section and it took some effort to clean it up
&lt;@ulm&gt; in fact, it was added to the devmanual very late, when we went through the old devrel handbook and collected all the forgotten things
&lt;@ulm&gt; and now you've filed a PR that "intends to state the obvious (but possibly unwritten) rules we have around" which is similar to what we had done before :)
&lt;@ulm&gt; sam_: so, maybe it would be better to split it up and add it to more specific chapters
&lt;@ulm&gt; for example, EAPI stuff could go to ebuild-writing/eapi
&lt;@sam_&gt; I mostly agree but I also think it's useful to have a single list of responsibilities
&lt;@sam_&gt; we have a problem where people say "I didn't realise I needed to do X"
&lt;@sam_&gt; perhaps we can find homes for some of the entries but maybe not all
&lt;@sam_&gt; I'll need to find the old section to compare with the content then
&lt;@ulm&gt; the stabilisation things could obviously be added to "Keywording and Stabilization" which is a toplevel chapter
&lt;@ulm&gt; old section?
&lt;@sam_&gt; you mentioned there was a previous section of this type
&lt;@sam_&gt; i'd like to compare to see if the old content was more amenable to being split (as it evidently was) and compare
&lt;@ulm&gt; sam_: you'd have to go through commit history of ebuild-maintenance/text.xml
&lt;@ulm&gt; and ebuild-writing/ebuild-maintenance/text.xml before that
&lt;@ulm&gt; it's messy :)
&lt;@ulm&gt; so yeah, it's basically three things you're talking about, EAPI, stabilisation, and pkgcheck/ci
&lt;@ulm&gt; which aren't much related, except that they're all "dev responsibilities". but we could say that about the whole devmanual content :)

**Comment by thesamesam** [comment]
Further context too for me to take into account on revisions:
```
[2021-02-17T15:42:45+0000] <@sam_> ulm: I'm not fully sure what to say
[2021-02-17T15:42:57+0000] <@sam_> ulm: We can add them to the individual pages, but I'm worried this will make it easier for them to get lost
[2021-02-17T15:43:10+0000] <@sam_> the idea was to list things which you need to actively do, rather than things to consider when e.g. writing a new ebuild
[2021-02-17T15:49:40+0000] <@antarus> I'd consider just making a section for responsibilities
[2021-02-17T15:49:50+0000] <@antarus> many of them are...principles? perhaps?
[2021-02-17T15:49:54+0000] <@antarus> more than specific guidelines
[2021-02-17T15:50:25+0000] <@antarus> I mean, maybe first, start by making a page with all of the responsibilities on it
[2021-02-17T15:50:32+0000] <@antarus> and then try to figure out where they all might go?
[2021-02-17T15:51:03+0000] <@antarus> e.g. is it a 'dev responsibility' not to cause excessive rebuilds?
[2021-02-17T15:51:14+0000] <@antarus> is it a 'dev responsibility' not to cause excessive metadata regen?
[2021-02-17T15:51:26+0000] <@antarus> is it a 'dev responsibility' not to cause nasty depgraphs for end users?
[2021-02-17T15:53:25+0000] <@sam_> i think that sounds like a good plan
[2021-02-17T15:53:30+0000] <@sam_> start with that, go from there
```

**Comment by thesamesam** [comment]
Will return to this.

**Pull request closed without merging**
